### PR TITLE
Java 21 Migration

### PR DESCRIPTION
**Pull request by Sameersawant77**
Fixes https://github.com/javapathfinder/jpf-core/issues/575

Refactoring and Migrating the codebase to support Java 21
I have reduced test failures from 791 down to 9. I am currently cleaning up the remaining edge cases and will open a PR shortly.

Changes:
Gradle Upgrade
Native Peers
Internal Accessors
Virtual Threads and Threads Container Handling

image

### CODE REVIEW

File: src/classes/modules/java.base/java/lang/Class.java [hunk 1]
```diff
@@ -17,24 +17,19 @@
...
 import java.io.Serializable;
 import java.lang.annotation.Annotation;
-import java.lang.reflect.AnnotatedElement;
-import java.lang.reflect.Constructor;
-import java.lang.reflect.Field;
-import java.lang.reflect.GenericDeclaration;
-import java.lang.reflect.Method;
-import java.lang.reflect.Type;
-import java.lang.reflect.TypeVariable;
+import java.lang.reflect.*;
```
Comment (cyrille-artho) [inline src/classes/modules/java.base/java/lang/Class.java]: Please import all packages explicitly to facilitate seeing dependencies.

File: src/classes/modules/java.base/java/lang/Class.java [hunk 2]
```diff
@@ -57,14 +52,14 @@ public final class Class<T> implements Serializable, GenericDeclaration, Type, A
 
    // we init this on demand (from MJIEnv) since it's not used too often
   private static Annotation[] emptyAnnotations; // = new Annotation[0];
-
```
Comment (cyrille-artho) [inline src/classes/modules/java.base/java/lang/Class.java]: We would like to separate non-functional changes from code changes so we can merge this patch with other branches in the future, so please try to not change whitespace.

File: src/classes/modules/java.base/jdk/internal/misc/CDS.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+public class CDS {
+	public static void initializeFromArchive(Class<?> c){
+
+	}
+
+	public static long getRandomSeedForDumping(){
+		return 0L;
+	}
+}
```
Comment (cyrille-artho) [inline src/classes/modules/java.base/jdk/internal/misc/CDS.java]: Add newline.
